Include pressure gradient source term in GeoClaw into Riemann Solver

mandli opened the pull request:
Add the gradient of pressure into the Riemann solver calculation rather than in the source term splitting.  This has important advantages when considering strong pressure gradients but also leads to significant speeds up to codes that use pressure gradient forcing (e.g. storm surge).

The corresponding GeoClaw PR is clawpack/geoclaw#242.

rjleveque commented on src/rpn2_geoclaw.f:
```diff
@@ -57,14 +59,22 @@ subroutine rpn2(ixy,maxm,meqn,mwaves,maux,mbc,mx,
       logical rare1,rare2
 
+      ! Density used in pressure gradient calculation 
+      ! TODO - Should add this as a parameter to geoclaw_module
```
Why not do this now in clawpack/geoclaw#242?

mandli replied:
I was not certain if this should be a parameter in the Python as well.  It would not be difficult to add if that sounds good.

mandli replied:
I would also probably add the density of air given that comes up every once in awhile as well.

mandli commented:
I moved `rho` into the `geoclaw_module` along with a couple of other, similar parameters. 

I also removed the parameter in the multilayer code that stored the density in place of the new `geoclaw_module` parameter.  The only users that will actually be effected are those using the multilayer code.
